src/Kentico.Xperience.Shopify/Config/ShopifyIntegrationSettingsService.cs

```diff
@@ -10,22 +13,28 @@ namespace Kentico.Xperience.Shopify.Config
     internal class ShopifyIntegrationSettingsService : IShopifyIntegrationSettingsService
     {
         private readonly IProgressiveCache cache;
-        private readonly IOptionsMonitor<ShopifyConfig> monitor;
+        private readonly IOptionsMonitor<ShopifyConfig> shopifyConfigMonitor;
         private readonly IInfoProvider<IntegrationSettingsInfo> integrationSettingsProvider;
+        private readonly IOptionsMonitor<ShopifyWebsiteChannelConfigOptions> websiteChannelConfigMonitor;
+        private readonly IWebsiteChannelContext websiteChannelContext;
 
         public ShopifyIntegrationSettingsService(
             IProgressiveCache cache,
-            IOptionsMonitor<ShopifyConfig> monitor,
-            IInfoProvider<IntegrationSettingsInfo> integrationSettingsProvider)
+            IOptionsMonitor<ShopifyConfig> shopifyConfigMonitor,
+            IInfoProvider<IntegrationSettingsInfo> integrationSettingsProvider,
+            IOptionsMonitor<ShopifyWebsiteChannelConfigOptions> websiteChannelConfigMonitor,
+            IWebsiteChannelContext websiteChannelContext)
         {
             this.cache = cache;
-            this.monitor = monitor;
+            this.shopifyConfigMonitor = shopifyConfigMonitor;
```

Comment:
```suggestion
            this.shopifyConfig = shopifyConfigMonitor.CurrentValue;
```
You can set current value in constructor and use it right away (so you dont have to call it in every methods)

Same for ShopifyWebsiteChannelConfigOptions